File: app/src/main/java/org/opennms/arnet/api/model/Alarm.java
package org.opennms.arnet.api.model;

import java.util.Date;

public interface Alarm {

    enum Severity {
        INDETERMINATE,
        CLEARED,
        NORMAL,
        WARNING,
        MINOR,
        MAJOR,
        CRITICAL
    }

    String getReductionKey();

    Severity getSeverity();

    String getDescription();

    Date getLastUpdated();

    String getVertexId();

}

File: model/src/main/kotlin/org/opennms/arnet/model/Alarm.kt
package org.opennms.arnet.model

import java.util.*

enum class Severity {
    INDETERMINATE,
    CLEARED,
    NORMAL,
    WARNING,
    MINOR,
    MAJOR,
    CRITICAL
}

data class Alarm(
    val reductionKey: String,
    val severity: Severity,
    val description: String,
    val lastUpdated: Date,
    val vertexId: String
)